pkg/agent/openflow/client.go

```diff
@@ -1129,3 +1143,33 @@ func (c *client) SendIGMPQueryPacketOut(
 	packetOutObj := packetOutBuilder.Done()
 	return c.bridge.SendPacketOut(packetOutObj)
 }
+
+func (c *client) InstallTrafficControlMarkFlows(name string, sourceOFPorts []uint32, targetOFPort uint32, direction v1alpha2.Direction, action v1alpha2.TrafficControlAction) error {
+	flows := c.featurePodConnectivity.trafficControlMarkFlows(sourceOFPorts, targetOFPort, direction, action)
+	cacheKey := fmt.Sprintf("tc_%s", name)
+	c.replayMutex.RLock()
+	defer c.replayMutex.RUnlock()
+	return c.modifyFlows(c.featurePodConnectivity.tcCachedFlows, cacheKey, flows)
```

Comment:
Using modifyFlows, it might delete the original tc flows which are not included in the `flows` which is generated by this call. I think your target is to append these flows in the cache but not overwrite?

Reply:
Since a traffic rule could be updated, the original tc flows should be deleted and new flows should be installed. Flows in the cache should be also overwritten.

Reply:
So do you mean `InstallTrafficControlMarkFlows` is called for both add and update case? And `UninstallTrafficControlMarkFlows` is called only when delete the tc configuration?

Reply:
Yes